File: 0169-majority-element/169-majority-element.py
class Solution:
	def majorityElement(self, nums: list[int]) -> int:
		hashmap = {}

		for i in nums:
			hashmap[i] = hashmap.get(i, 0) + 1
			if hashmap[i] > len( nums )// 2:
				return i

		return -1

sol = Solution()

arr = [2,2,1,1,1,2,2]
cnt = sol.majorityElement( arr )
print(cnt)

File: 0169-majority-element/169-majority-element.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> hashmap = new HashMap<>();

		for ( int i = 0; i < nums.length; i++ ) {
            hashmap.put(nums[i], hashmap.getOrDefault(nums[i], 0) + 1);
			if ( hashmap.get(nums[i]) > nums.length/ 2 ) {
				return nums[i];
			}
		}
		return -1;
    }

	public static void main( String[] args ) {
		Solution sol = new Solution();
		int[] arr = {2,2,1,1,1,2,2};
		int cnt = sol.majorityElement( arr );
		System.out.println(cnt);
	}
}
